apluslms-yamlidator/apluslms_yamlidator/validator.py

```diff
@@ -214,9 +215,18 @@ def set_defaults(validator, properties, instance, schema):
                 instance.setdefault(property, subschema["default"])
         yield from validate_properties(validator, properties, instance, schema)
 
+    def python_type(validator, types, instance, schema):
+        if not isinstance(types, list):
+            types = [types]
+
+        if not any(instance.__class__.__name__ == type_ for type_ in types):
+            yield ValidationError(types_msg(instance, types))
+
     new = validators.extend(validator_class,
         # redefine 'properties' validator
-        validators={"properties" : set_defaults},
+        validators={
+            "properties" : set_defaults,
+            "python_type": python_type},
```

Comment:
prefer `pythonType` to match json schema style

Reply:
well these also use camel case: https://github.com/Julian/jsonschema/blob/master/jsonschema/_validators.py
but I guess I'll change is back to snake case to match our functions

Reply:
Yes. That library uses non pythonic way for naming the functions, so they are same as the strings in the schema files. Pretty acceptable as those functions are not exposed outside of the library.